event/models.py

```diff
@@ -246,3 +255,48 @@ def __str__(self):
 
     class Meta:
         ordering = ["position"]
+
+
+class Participant(models.Model):
+    first_name = models.CharField(max_length=255)
+    last_name = models.CharField(max_length=255)
+    email = models.EmailField(max_length=255)
+    academy = models.CharField(max_length=255, blank=True)
+    phone_number = models.CharField(
+        max_length=14,
+        validators=[
+            RegexValidator(
+                # makes sure the it is at least 10 and no more than 14
+                regex=r"^\+?[0-9]{10,14}$",
+                message=(
+                    "Phone number must be 10 digit",
+                    "long and  Up to 14 digits allowed including +977",
+                ),
+            )
+        ],
+    )
+    events = models.ForeignKey(
+        Event,
+        related_name="participants",
+        on_delete=models.CASCADE,
+        blank=False,
+        null=False,
+    )
+    age = models.IntegerField(null=False, blank=False)
+
+    def clean(self):
+        """
+        Do not allow saving the participant if the event is full.
+        """
+        if self.events.participants.count() >= self.events.max_capacity:
+            raise ValidationError("Event is full")
+        if not self.events:
+            raise ValidationError("Event is required")
```

Comment:
```suggestion
        if not self.events:
            raise ValidationError("Event is required")
        if self.events.participants.count() >= self.events.max_capacity:
            raise ValidationError("Event is full")
```

Placing ```if not self.events``` first prevents potential AttributeError when accessing ```self.events.participants.count()```